flink-ai-flow/ai_flow/api/workflow_operation.py

```diff
@@ -78,25 +86,26 @@ def submit_workflow(workflow_name: Text = None,
     """
     if default_graph().is_empty():
         raise EmptyGraphException("Cannot submit empty graph")
-    call_path = os.path.abspath(sys._getframe(1).f_code.co_filename)
-    project_path = os.path.abspath(project_description().project_path)
-    # length /python_codes/ is 14; length .py is 3
-    entry_module_path = call_path[len(project_path)+14:-3].replace('/', '.')
+    entry_module_path = project_description().get_workflow_entry_module(workflow_name=workflow_name)
     namespace = project_config().get_project_name()
     translator = get_default_translator()
     workflow = translator.translate(graph=default_graph(), project_desc=project_description())
+    apply_full_info_to_workflow(entry_module_path, workflow)
     default_graph().clear_graph()
-    for job in workflow.jobs.values():
-        _register_job_meta(workflow_id=workflow.workflow_id, job=job)
-    _set_entry_module_path(workflow, entry_module_path)
-    _upload_project_package(workflow)
     return proto_to_workflow(get_ai_flow_client()
                              .submit_workflow_to_scheduler(namespace=namespace,
                                                            workflow_json=json_utils.dumps(workflow),
                                                            workflow_name=workflow_name,
                                                            args=args))
 
 
+def apply_full_info_to_workflow(entry_module_path, workflow):
```

Comment:
Should the `apply_full_info_to_workflow` method be private? Users don't need to invoke the method.